assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -155,6 +169,20 @@ const baseActions = {
 			}
 
 			yield propertyActions.selectProperty( uaProperty?.id, uaProperty?.internalWebPropertyId ); // eslint-disable-line sitekit/acronym-case
+
+			if ( ! isFeatureEnabled( 'ga4setup' ) ) {
+				return;
+			}
+
+			registry.dispatch( STORE_NAME ).setPrimaryPropertyType( PROPERTY_TYPE_UA );
+
+			const ga4MatchProperty = registry.dispatch( MODULES_ANALYTICS_4 ).matchAndSelectProperty( accountID, GA4_PROPERTY_CREATE );
+			const ga4Property = yield Data.commonActions.await( ga4MatchProperty );
```

Comment:
Is this await needed? The returned property comes from `matchPropertyByURL` which already uses this on the returned value.

Reply:
Yes, this `await` is needed because we need to wait till the `matchAndSelectProperty` generator action (which returns a promise) finishes.

Reply:
I see, it's because we use `registry.dispatch` here which always returns a promise. A bit unsightly but unavoidable for now. This will be much cleaner after #1769 as we'll be able to simply `yield dispatch( store, ... )`. Thanks 👍